modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -513,8 +513,12 @@ class SettingsImpl @Inject constructor(
         return getBoolean(Settings.PREFERENCE_AUTO_SHOW_PLAYED, false)
     }
 
-    override fun canDuckAudioWithNotifications(): Boolean {
-        return sharedPreferences.getBoolean(Settings.PREFERENCE_OVERRIDE_AUDIO, false)
+    override fun getPlayOverNotification(): Int {
+        val value = sharedPreferences.getString(
+            Settings.PREFERENCE_NOTIFICATION_AUDIO,
+            Settings.PREFERENCE_NOTIFICATION_AUDIO_DEFAULT
+        ) ?: Settings.PREFERENCE_NOTIFICATION_AUDIO_DEFAULT
+        return Integer.parseInt(value)
```

Comment:
Yea great pickup and much more user friendly than a user losing their previous preference, I made an adjustment like so:

```
  override fun getPlayOverNotification(): PlayOverNotificationSetting {
        val value = sharedPreferences.getString(Settings.PREFERENCE_NOTIFICATION_AUDIO, null) ?: legacyPlayOverNotification()
        return PlayOverNotificationSetting.fromPreferenceString(value)
    }

    private fun legacyPlayOverNotification(): String {
        if (sharedPreferences.getBoolean(Settings.PREFERENCE_OVERRIDE_AUDIO, false)) {
            return PlayOverNotificationSetting.ALWAYS.preferenceInt.toString()
        }

        return PlayOverNotificationSetting.NEVER.preferenceInt.toString()
    }
```

So basically, check if they have a preference set, if not check then check if the legacy preference is set to true then return `Always` otherwise return `Never`.

I gave it a quick test on my device and it worked well for your test. Happy to modify if though if it can be improved further.